lib/Sema/TypeCheckConcurrency.cpp

```diff
@@ -4436,6 +4436,8 @@ ActorIsolation ActorIsolationRequest::evaluate(
         (isolation != ActorIsolation::NonisolatedUnsafe)) {
       auto *classDecl = var->getDeclContext()->getSelfClassDecl();
       const bool isActorType = classDecl && classDecl->isAnyActor();
+      const bool isPreconcurrencyUnspecifiedIsolation =
+          var->preconcurrency() && isolation.isUnspecified();
```

Comment:
I am interested in feedback as to whether here there is any meaningful difference between testing for `var->preconcurrency()` vs `isolation.preconcurrency()`?

Reply:
If I'm reading everything correctly ActorIsolation produced by `getIsolationFromAttributes` would not have "preconcurrency" bit set for explicitly nonisolated or unspecified. It seems better to use `var->preconcurrency()` just like you did in this case.

Reply:
Thank you. It turned out that my logic was more deeply flawed in that those are checks for `@preconcurrency` as an attribute on a declaration, but do not capture when `@preconcurrency import` is used. I just pushed a replacement commit that undoes most of the previous one and appropriately manages diagnostic level changes based on `@preconcurrency import` usage instead.

Reply:
I've been bumping into issues with `Decl::preconcurrency` lately. It does not include `@preconcurrency` attributes that are inferred. It basically represents explicit `@preconcurrency` annotations + it's always true for imported declarations from ObjC.

If there are cases where `ActorIsolation` does not preserve the preconcurrency bit, we should fix that.